Add blog post for FSGroup on mount (#37621)
kubernetes/website: content/en/blog/_posts/2022-12-23-fsgroup-on-mount.md

---
layout: blog

> title: "Kubernetes 1.26: Providing pod's FSGroup to CSI drivers during mount"

Reviewer: ```suggestion
title: "Kubernetes 1.26: Support for Passing Pod fsGroup to CSI Drivers At Mount Time"
```
?

the `fsGroup` information to all the files in the volume.

It should be noted that in the example above the kubelet refrains from directly
applying the permission changes into the files and directories in that volume files. 
Additionally, two policy definitions no longer have an effect: neither
`.spec.fsGroupPolicy` for the CSIDriver object, nor
`.spec.securityContext.fsGroupChangePolicy` for the Pod.

For more details about the inner workings of this feature, check out the
[enhancement proposal](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/2317-fsgroup-on-mount/)
and the [CSI Driver `fsGroup` Support](https://kubernetes-csi.github.io/docs/support-fsgroup.html)
in the CSI developer documentation.

## Why is it important?

Without this feature, applying the fsGroup information to files is not possible in certain storage environments.

For instance, Azure File does not support a concept of POSIX-style ownership and permissions
of files. The CSI driver is only able to set the file permissions at the volume level.

## How do I use it?

> This feature should be mostly transparent to users. If you maintain a CSI driver that should

Reviewer 2: nit
```suggestion
This feature should be mostly transparent to users. If you maintain a CSI driver that
```

Reviewer 2: As an addition, if all CSI drivers don't support this, do we want to link to a supported list?

Reviewer: CSI drivers are third party software. Unless https://landscape.cncf.io/ covers them, there probably isn't any definitive list.

> support this feature, read

Reviewer 2: ```suggestion
supports this feature, read
```

Reviewer: “should support” looks idiomatic to me.
